react-components/src/components/Minicart/cartInitializer.js

```diff
@@ -55,7 +57,10 @@ const CartInitializer = props => {
                 });
                 dispatch({ type: 'open' });
                 dispatch({ type: 'beginLoading' });
-                return addItem({
+
+                let addItemFunc = ev.detail.virtual ? addVirtualItem : addItem;
```

Comment:
Hmmm, shouldn't we have that `virtual` attribute per item added to the cart? What if we have a grouped product composed of two virtual products and one normal product? Not sure if this is realistic, but it would be better to support that, WDYT?

Reply:
I will try if a grouped product can be composed of virtual and physical products.

We would need to split the query then however into `addSimpleProductsToCart` and `addVirtualProductsToCart` mutations.

Reply:
Not sure if this works OOTB in Magento, but I've seen countless of "Guitar, amp and 1yr subscription to online course" offers, so it's a valid business case.

Reply:
I created a follow up JIRA for this and will work on it after the release :)